### DISCUSSION TITLE
接收消息

### DISCUSSION THREAD

**Discussion by Xiaoxusheng** [Mrs4s/go-cqhttp#1879] [General]
你好。websocket怎么接收消息，能有例子代码吗

**Comment by SlimeNull** [comment]
和 HTTP 上报是一样的, 在文章中提到了 HTTP 上报的包体结构, 是一个 JSON, WebSocket 也是如此, 你会从 go-cqhttp 那里收到一个包含上报信息的 JSON, 通过判断 post_type 来判断这个上报是什么类型.

当然, 你也可以参考 [EleCho.GoCqHttpSdk](https://github.com/OrgEleCho/EleCho.GoCqHttpSdk) 中对正向 WebSocket 的实现

**Comment by SlimeNull** [comment]
另外, 在 `api` 一节中, 也有写明使用 WebSocket 时的数据内容
